### ISSUE TITLE
Crashing in NetworkManager.java line 8 when ConnectionException occured

### ISSUE THREAD

**Issue by raymondctc** [Instabug/Instabug-Android#323]
### Steps to Reproduce the Problem
n/a

### Expected Behavior
No crashing

### Actual Behavior
Crashed

### Stack trace
```
Fatal Exception: tc8: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | java.net.ConnectException: Failed to connect to api.instabug.com/0.0.0.0:443
       at io.reactivex.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:367)
       at io.reactivex.internal.operators.completable.CompletableMergeIterable$MergeCompletableObserver.onError(CompletableMergeIterable.java:124)
       at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.onError(CompletableSubscribeOn.java:74)
       at io.reactivex.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onError(CompletableAndThenCompletable.java:62)
       at io.reactivex.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onError(CompletableAndThenCompletable.java:62)
       at io.reactivex.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onError(CompletablePeek.java:95)
       at io.reactivex.internal.operators.observable.ObservableIgnoreElementsCompletable$IgnoreObservable.onError(ObservableIgnoreElementsCompletable.java:61)
       at io.reactivex.internal.operators.observable.ObservableRetryPredicate$RepeatObserver.onError(ObservableRetryPredicate.java:79)
       at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.tryOnError(ObservableCreate.java:84)
       at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onError(ObservableCreate.java:72)
       at com.instabug.library.network.NetworkManager$a.subscribe(NetworkManager.java:104)
       at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
       at io.reactivex.Observable.subscribe(Observable.java:12284)
       at io.reactivex.internal.operators.observable.ObservableRetryPredicate$RepeatObserver.subscribeNext(ObservableRetryPredicate.java:112)
       at io.reactivex.internal.operators.observable.ObservableRetryPredicate.subscribeActual(ObservableRetryPredicate.java:41)
       at io.reactivex.Observable.subscribe(Observable.java:12284)
       at io.reactivex.internal.operators.observable.ObservableIgnoreElementsCompletable.subscribeActual(ObservableIgnoreElementsCompletable.java:31)
       at io.reactivex.Completable.subscribe(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
       at io.reactivex.Completable.subscribe(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
       at io.reactivex.Completable.subscribe(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
       at io.reactivex.Completable.subscribe(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
       at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
       at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
       at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)
```

```
Caused by java.net.ConnectException: Failed to connect to api.instabug.com/0.0.0.0:443
       at com.android.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:147)
       at com.android.okhttp.internal.io.RealConnection.connect(RealConnection.java:116)
       at com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:186)
       at com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:128)
       at com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:97)
       at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:289)
       at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:232)
       at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:465)
       at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:131)
       at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:262)
       at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:219)
       at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:30)
       at com.google.firebase.perf.network.zzf.getOutputStream(zzf.java:67)
       at com.google.firebase.perf.network.zzc.getOutputStream(zzc.java:13)
       at com.instabug.library.network.NetworkManager.connectWithNormalType(NetworkManager.java:8)
       at com.instabug.library.network.NetworkManager.access$100(NetworkManager.java:1)
       at com.instabug.library.network.NetworkManager$a.subscribe(NetworkManager.java:20)
       at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
       at io.reactivex.Observable.subscribe(Observable.java:12284)
       at io.reactivex.internal.operators.observable.ObservableRetryPredicate$RepeatObserver.subscribeNext(ObservableRetryPredicate.java:112)
       at io.reactivex.internal.operators.observable.ObservableRetryPredicate.subscribeActual(ObservableRetryPredicate.java:41)
       at io.reactivex.Observable.subscribe(Observable.java:12284)
       at io.reactivex.internal.operators.observable.ObservableIgnoreElementsCompletable.subscribeActual(ObservableIgnoreElementsCompletable.java:31)
       at io.reactivex.Completable.subscribe(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
       at io.reactivex.Completable.subscribe(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
       at io.reactivex.Completable.subscribe(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
       at io.reactivex.Completable.subscribe(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
       at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
       at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
       at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)
```

### Instabug integration code
```
        Instabug.Builder(application, application.getString(R.string.instabug_app_id))
                .setInvocationEvents(InstabugInvocationEvent.NONE)
                .build()
        Replies.setPushNotificationState(Feature.State.ENABLED)
        Instabug.setDebugEnabled(!AppBuildConfig.App.IS_RELEASE)
        Instabug.setPrimaryColor(ResourcesCompat.getColor(resources, R.color.under9_theme_black, null))
        // Arguments:initialScreenshot, extraScreenshot, galleryImage & ScreenRecording
        BugReporting.setAttachmentTypesEnabled(false, true, true, true)
        Surveys.setShouldShowWelcomeScreen(true)
        Chats.setState(Feature.State.DISABLED)
        CrashReporting.setState(Feature.State.DISABLED)
```

### SDK Version
9.1.6

### Android Version
All

### Device Model
n/a

**Comment by mzelzoghbi** [comment]
Hi @raymondctc , Thanks for reporting this.
Are you adding the integration code to the application class or in the activity class? Can you also tell us how frequently is this crash happening?

Thanks

**Comment by raymondctc** [comment]
> Hi @raymondctc , Thanks for reporting this.
> Are you adding the integration code to the application class or in the activity class? Can you also tell us how frequently is this crash happening?
> 
> Thanks

Hi @mzelzoghbi, the integration code is inside application class. We are not able to reproduce this issue on our devices, but seeing them happen on our users. Our new app version was released on 16th Jul, this was the 2nd top issue on Crashlytics after upgrading to 9.1.6. We then reverted it back to 9.1.0 and seeing much much less.

**Comment by Muhammadyoussef** [comment]
Hi @raymondctc,

This issue has been fixed in our major release `v10.0.0`. Please give it a go and feel free to reopen this issue if it still occurs
